### PR TITLE
Fix that GoogleTotpProvider ID is can not override.

### PR DESCRIPTION
**Pull request by yu-ishicawa**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://github.com/mohiva/play-silhouette/blob/master/CONTRIBUTING.md)?
* [ ] Have you added copyright headers to new files?
* [ ] Have you suggest documentation edits?
* [ ] Have you added tests for any changed functionality?

## Purpose

I want to create a class that inherits GoogleTotpProider and inject id dynamically.

```
class CustomTotpProvider @Inject()(
    override val id: String,
    passwordHasherRegistry: PasswordHasherRegistry
)(implicit ec: ExecutionContext)
    extends GoogleTotpProvider(passwordHasherRegistry) {
  ...
}
```

However, the authenticate method always returns the same ID ("googleTotp").

```
  def authenticate(sharedKey: String, verificationCode: String): Future[Option[LoginInfo]] = {
    Future(
      if (isVerificationCodeValid(sharedKey, verificationCode)) {
        Some(LoginInfo(ID, sharedKey))
                       ^^
    ...
```

### CODE REVIEW

**Comment by bravegag** [comment]
Hi @yu-ishicawa, I worked in the `GoogleTotpProvider` implementation. This is not intended to be extended from. If you have your own form of TOTP or 2FA then you should make a sibling (replicate the `GoogleTotpProvider` implementation). Unfortunately we were discouraged from creating a common base class for all types of TOTP implementations. See the discussion here:
https://github.com/mohiva/play-silhouette/pull/557#issuecomment-506626712

In other words, it doesn't make sense what you are trying to do. 

The ID you are trying to use exists to uniquely identify the `GoogleTotpProvider` as provider and it's  not intended for overriding, in fact, it should be final in this implementation.

@akkie See what I meant, it is all about UX, we wanted the base `TotpProvider` :)

**Comment by akkie** [comment]
@bravegag I have to disagree with you here. I don't think this is about changing the functionality of the provider. It's about of changing the ID of the provider, for whatever reason.
One use case would be to use providers with different configurations.

From a logical standpoint, I think this PR is valid and fixes an issue. Code in the provider should use the `id` method and not the constant. I think all other providers do this the same way.

**Comment by yu-ishicawa** [comment]
> @bravegag I have to disagree with you here. I don't think this is about changing the functionality of the provider. It's about of changing the ID of the provider, for whatever reason.
> One use case would be to use providers with different configurations.

Exactly. What I wanted to do was simply to use two different IDs, not to change the behavior of GoogleTotpProvider.

I have defined the CustomTotpProvider initialization as follows:

```
  @Provides
  def provideTotpProvider(
      @Named("googleTotp-provider-id") providerId: String,
      passwordHasherRegistry: PasswordHasherRegistry
  ): GoogleTotpProvider = {
    new CustomTotpProvider(providerId, passwordHasherRegistry)
  }
```

A module defines the ID to inject as follows:

```
  @Provides @Named("googleTotp-provider-id")
  def provideGoogleTotpProviderId: String = "googleTotp-a"
```

The other is defined as follows:

```
  @Provides @Named("googleTotp-provider-id")
  def provideGoogleTotpProviderId: String = "googleTotp-b"
```

Thank you for the discussion.

**Comment by akkie** [comment]
@yu-ishicawa Thanks :+1:

**Comment by akkie** [comment]
@yu-ishicawa I've published version 6.1.1 with your fix. Thanks again
